### PR TITLE
Setup production build

### PR DESCRIPTION
**Pull request by jkhabra**
- Upload docker images to docker-hub

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
+on:
+  push:
+    branches:
+      - feature/setup-production-build
```
Comment (bitspook) [inline .github/workflows/deploy.yml]: The target branch should be `master` now.

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    name: Build GATEWAY DockerImage
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v1
+
+    - name: Login to DockerHub Registry
+      run: echo $DOCKER_PASSWORD | docker login -u $DOCKER_USER --password-stdin
+      env:
+        DOCKER_USER: ${{ secrets.DOCKER_USER }}
+        DOCKER_PASSWORD: ${{ secrets.DOCKER_PASSWORD }}
```
Comment (bitspook) [inline .github/workflows/deploy.yml:55]: @chahalinder0007 This is how you can put variables/keys in Github settings, and use them in workflow files and avoid dropping the passwords here.

**Review by bitspook — CHANGES_REQUESTED** [review]
Looks good! Just one small change requested.

File: .github/workflows/deploy.yml (hunk 2 above)
Comment (chahalinder0007) [reply .github/workflows/deploy.yml:55]: @channikhabra I know just too lazy to do that :P

**Review by bitspook — APPROVED** [review]

**Pull request merged**